test/parse-test.js

```diff
@@ -107,71 +120,121 @@ describe("dateFormat.parse", function() {
-      verifyDate(date, { hours: 3, minutes: 23 });
+      it("works with hh:mm O (-200)", function() {
+        var date = dateFormat.parse("hh:mm O", "05:23 -200");
+        verifyDate(date, { hours: 7, minutes: 23 });
+      });
 
-      date = dateFormat.parse("hh:mm O", "05:23 +0930");
-      verifyDate(date, { hours: 14, minutes: 53 });
+      it("works with hh:mm O (+0930)", function() {
+        var date = dateFormat.parse("hh:mm O", "05:23 +0930");
+        verifyDate(date, { hours: 19, minutes: 53 });
+      });
     });
   });
 
   describe("with a date formatted by this library", function() {
-    var testDate = new Date();
-    testDate.setUTCFullYear(2018);
-    testDate.setUTCMonth(8);
-    testDate.setUTCDate(13);
-    testDate.setUTCHours(18);
-    testDate.setUTCMinutes(10);
-    testDate.setUTCSeconds(12);
-    testDate.setUTCMilliseconds(392);
-
-    it("should format and then parse back to the same date", function() {
-      dateFormat
-        .parse(
-          dateFormat.ISO8601_WITH_TZ_OFFSET_FORMAT,
-          dateFormat(dateFormat.ISO8601_WITH_TZ_OFFSET_FORMAT, testDate)
-        )
-        .should.eql(testDate);
+    describe("should format and then parse back to the same date", function() {
+      function testDateInitWithUTC() {
+        var td = new Date();
+        td.setUTCFullYear(2018);
+        td.setUTCMonth(8);
+        td.setUTCDate(13);
+        td.setUTCHours(18);
+        td.setUTCMinutes(10);
+        td.setUTCSeconds(12);
+        td.setUTCMilliseconds(392);
+        return td;
+      }
+
+      function testDateInitWithLocal() {
+        var td = new Date();
+        td.setFullYear(2018);
+        td.setMonth(8);
+        td.setDate(13);
+        td.setHours(18);
+        td.setMinutes(10);
+        td.setSeconds(12);
+        td.setMilliseconds(392);
+        return td;
+      }
+
+      xit("works with ISO8601_WITH_TZ_OFFSET_FORMAT", function() {
+        // For this test case to work, the date object must be initialized with
+        // UTC timezone
+        var td = testDateInitWithUTC();
+        var d = dateFormat(dateFormat.ISO8601_WITH_TZ_OFFSET_FORMAT, td);
+        dateFormat.parse(dateFormat.ISO8601_WITH_TZ_OFFSET_FORMAT, d)
+          .should.eql(td);
+      });
 
-      dateFormat
-        .parse(
-          dateFormat.ISO8601_FORMAT,
-          dateFormat(dateFormat.ISO8601_FORMAT, testDate)
-        )
-        .should.eql(testDate);
+      // The followings test cases are invalid and would only work when running
+      // them with OS timezone set to UTC.  ISO8601 string is local time with 
+      // timezone info.  Essentially, when the ISO8601 string is converted to
+      // string without timezone, the timezone information is lost and not
+      // recoverable.
+      // These use cases will produce unpriditable results.  Use any of them if 
+      // you'd like to gamble or to make rest of your life miserable. 
+      xit("works with ISO8601_FORMAT", function() {
```

Comment:
I tried to make it work but it won't work for all cases.  As it is currently written, when parsing the date from string, we don't know for sure if timezone is provided by the input or by the missingValueDate.  If we use UTC to manipulate the date/time/hour/minute and the reference date is local, then you will get incorrect result at daylight saving time change.  Even if it uses non-UTC setter to manipulate those values, it can also land in the gap hour or dup hour.  In either case, it won't provide consistent result and will vary depend on the OS timezone and the date/time value.

Reply:
Yep, ok. I'll merge and see if I can get anything to work with your changes. If not, then I'll remove the tests.